hack/generate-manifest.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--version <antrea version>] [--help|-h]
+Generate a YAML manifest for Antrea using Kustomize and print it to stdout.
+        --version                     Antrea version for use for manifest generation
+        --help, -h                    Print this message and exit
+
+This tool uses kustomize (https://github.com/kubernetes-sigs/kustomize) to generate manifests for
+Antrea. You can set the KUSTOMIZE environment variable to the path of the kustomize binary you want
+us to use. Otherwise we will download the appropriate version of the kustomize binary and use
+it (this is the recommended approach since different versions of kustomize may create different
+output YAMLs)."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+MODE="dev"
+ANTREA_VERSION="main"
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --version)
+    ANTREA_VERSION="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+source $THIS_DIR/verify-kustomize.sh
+
+if [ -z "$KUSTOMIZE" ]; then
+    KUSTOMIZE="$(verify_kustomize)"
+elif ! $KUSTOMIZE version > /dev/null 2>&1; then
+    echoerr "$KUSTOMIZE does not appear to be a valid kustomize binary"
+    print_help
+    exit 1
+fi
+
+# Get Antrea repository
+ANTREA_DIR=$(mktemp -d /tmp/antrea.XXXXXXX)
```

Comment:
out of curiosity, is there a logic for the suffix length or are we just typing 'X' until we've had enough?

Reply:
mktemp will generate a file (or a dir in this case as -d is specified) with a name that matches the template. 'X'es are replaced with random characters - the length of the X string determines the length of the random string.